mint/install-ruby/mint-leaf.yml

```diff
@@ -11,15 +11,67 @@ parameters:
     required: false
 
 tasks:
-  - key: install-ruby-install
+  - key: install-ruby
     run: |
-      sudo apt-get update
-      $MINT_LEAF_PATH/bin/install-ruby-install
-      sudo apt-get clean
+      set -eou pipefail
 
-  - key: install-ruby
-    use: install-ruby-install
-    run: $MINT_LEAF_PATH/bin/install-ruby
+      # ensure we have a ruby version
+      if [ "$RUBY_VERSION" = "" ]; then
+        if [ "$RUBY_VERSION_FILE" = "" ]; then
+          >&2 echo "RUBY_VERSION or RUBY_VERSION_FILE must be set."
+          exit 2
+        fi
+
+        RUBY_VERSION="$(cat "$RUBY_VERSION_FILE")"
+        if [ "$RUBY_VERSION" = "" ]; then
+          >&2 echo "$RUBY_VERSION_FILE did not specify a Ruby version."
+          exit 2
+        fi
+      fi
+
+      # Find the prebuilt Ruby
+      echo "Looking for Ruby $RUBY_VERSION..."
+      escaped_version=$(echo "$RUBY_VERSION" | sed "s/\./\\\\./g") # 3.2.2 => 3\.2\.2
+      version_line=$(cat $MINT_LEAF_PATH/known-rubies.csv | grep "^$escaped_version,")
+      if [[ -z "$version_line" ]]; then
+        >&2 echo "Ruby $RUBY_VERSION is not available in this version of mint/install-ruby."
+        >&2 echo "Try upgrading to the most recent version of this leaf. If the version is still not available, file an issue at https://github.com/rwx-research/mint-leaves/issues/new."
+        exit 3
+      fi
```

Comment:
You could use `$MINT_ERRORS` here! For example https://github.com/rwx-research/mint/blob/cee77a920b879c6ba9ea9fead2132d9d76dc424b/packages/bigtest/errors.test.bigtest.ts#L23-L28

Reply:
Oh nice! I'll switch this over